### PR TITLE
Remove symbols from vault binary

### PR DESCRIPTION
**Pull request by BharathB23**
An optional Makefile target rule(dev-strip) to remove symbols from vault binary to make it secure.

### CODE REVIEW

**Comment by jefferai** [comment]
Please remove the Makefile part of this. The rest looks fine.

**Comment by BharathB23** [comment]
But how to set VAULT_STRIP_BINARIES, by using it as an environment variable?

**Comment by jefferai** [comment]
Yes, that's all that the Makefile is doing anyways.

**Comment by BharathB23** [comment]
Where should I capture about VAULT_STRIP_BINARIES, any document or README should be modified?

**Comment by jefferai** [comment]
It doesn't need to be documented.

**Comment by BharathB23** [comment]
Have removed the Makefile changes.

File: scripts/build.sh [hunk 1]
```diff
@@ -42,12 +42,16 @@ if [ "${VAULT_DEV_BUILD}x" != "x" ]; then
     XC_ARCH=$(go env GOARCH)
     XC_OSARCH=$(go env GOOS)/$(go env GOARCH)
 fi
+# If VAULT_STRIP_BINARIES env variable is set, then remove symbols from vault binary.
+if [[ "${VAULT_STRIP_BINARIES}x" != "x" ]]; then
+    LD_FLAGS="-s -w "
```
Comment (jefferai) [inline scripts/build.sh]: This will override any existing LD_FLAGS.

**Comment by BharathB23** [comment]
LD_FLAGS, was locally defined and used, override shouldn't happen unless someone reused the same var.
Changes made and committed to avoid it.

**Comment by BharathB23** [comment]
Sorry closed it by mistake, re-opening.

**Comment by jefferai** [comment]
If LD_FLAGS is now being passed into the build command, then there's no need for VAULT_STRIP_BINARIES at all, because you can just set the LD_FLAGS env var.

**Comment by BharathB23** [comment]
Please let me know, how to proceed.

**Comment by jefferai** [comment]
Keep only the last change (adding LD_FLAGS) and I'll merge it.

**Comment by BharathB23** [comment]
Removed additional VAULT_STRIP_BINARIES check.

**Pull request merged**